23.0.0.1 staging (#2875)
OpenLiberty/blogs: posts/2023-02-07-23.0.0.1.adoc

Check out link:{url-prefix}/blog/?search=release&search!=beta[previous Open Liberty GA release blog posts].

[#run]

== Run your apps using 23.0.0.1

If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:

[source,xml]
----
<dependency>
    <groupId>io.openliberty</groupId>
    <artifactId>openliberty-runtime</artifactId>
    <version>23.0.0.1</version>
    <type>zip</type>
</dependency>
----

Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:

[source,gradle]
----
dependencies {
    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[23.0.0.1,)'
}
----

Or if you're using Docker:

[source]
----
FROM open-liberty
----

Or take a look at our link:{url-prefix}/downloads/[Downloads page].

[link=https://stackoverflow.com/tags/open-liberty]
image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/24069
// Contact/Reviewer: ReeceNana,cbridgha
// // // // // // // // 
[#eplv2]
== Open Liberty is now licensed under EPLv2

> On December 7th, 2022, we published the link:{url-prefix}/blog/2022/12/07/relicense-eplv2-proposal.html[Proposal to relicense Open Liberty to EPLv2] blog post to solicit feedback from the community regarding changing Open Liberty's license from EPLv1 to EPLv2.  Having received no negative feedback to the proposal, we've moved forward with the change.  As a result, Open Liberty 23.0.0.1 has been released under the EPLv2 license.  For more information regarding the reasoning and implications of moving from EPLv1 to EPLv2, refer to the link:{url-prefix}/blog/2022/12/07/relicense-eplv2-proposal.html[blog post].

Reviewer: ```suggestion
On December 7th, 2022, we published the link:{url-prefix}/blog/2022/12/07/relicense-eplv2-proposal.html[Proposal to relicense Open Liberty to EPLv2] blog post to solicit feedback from the community about changing Open Liberty's license from EPLv1 to EPLv2.  Having received no negative feedback to the proposal, we've moved forward with the change.  As a result, Open Liberty 23.0.0.1 is released under the EPLv2 license.  For more information about the reasoning behind and implications of moving from EPLv1 to EPLv2, refer to the link:{url-prefix}/blog/2022/12/07/relicense-eplv2-proposal.html[proposal blog post].
```

// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 

[#bugs]
== Notable bugs fixed in this release

We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A23001+label%3A%22release+bug%22[full list of bugs fixed in 23.0.0.1].

* link:https://github.com/OpenLiberty/open-liberty/issues/23885[Use mininum jdkSourceLevel of 1.8 for JDK 20+]
+
In JDK 20, `javac` has removed support for Java 1.7 compiling. This change causes the following errors in JSP when `jdkSourceLevel="17"` (Java version 1.7) is specified:
+
[source]
----
com.ibm.ws.jsp.JspCoreException: JSPG0049E: /xxxxxxxxx.jsp failed to compile :
error: Source option 7 is no longer supported. Use 8 or later.
----
+

> The jdkSourceLevel specified by the user will be overrided to "18" if a lower level is specified, when Java 20 or higher is the runtime.

Reviewer: ```suggestion
When Java 20 or higher is the runtime, the `jdkSourceLevel` specified by the user is overwritten to `18` if a lower level is specified.
```

> After the fix, when Java 20 or higher is used and a `jdkSourceLevel` is set to "17" or lower, the value will be overwritten to "18".

Reviewer: ```suggestion
After the fix, when Java 20 or higher is used and a `jdkSourceLevel` is set to `17` or lower, the value is overwritten to `18`.
```

* link:https://github.com/OpenLiberty/open-liberty/issues/23883[Default keystore file not getting detected on file monitoring]
+

> An issue was present which prevented the default keystore file from being reloaded when a change occurred if `updateTrigger` used `mbean` (the default value).

Reviewer: ```suggestion
An issue was present that prevented the default keystore file from being reloaded when a change occurred if the `updateTrigger` attribute used `mbean` (the default value).
```

+
The issue has been resolved and the file monitoring now properly occurs for this scenario, causing the file to be reloaded as expected.

* link:https://github.com/OpenLiberty/open-liberty/issues/23782[JDBCDriverService; issue with Boolean parameters]
+

> A change in the Microsoft JDBC driver has caused a regression when boolean values are used for attributes in `properties.microsoft.sqlserver`.  This causes an error like the following to occur:

Reviewer: ```suggestion
A change in the Microsoft JDBC driver has caused a regression when boolean values are used for attributes in the `properties.microsoft.sqlserver` property. This change causes an error similar to the following example:
```

+
[source]
----
com.ibm.ws.jdbc.internal.JDBCDriverService W DSRA8021W: Warning: error setting 'encrypt'=true: java.lang.IllegalArgumentException: argument type mismatch
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
----
+
A change has been delivered that converts the `boolean` value to the now expected `String`, resolving the issue.

* link:https://github.com/OpenLiberty/open-liberty/issues/23771[IndexOutOfBoundsException can occur during a resource outage.]
+

> A `java.lang.IndexOutOfBoundsException` can occur during a back end resource outage. During an outage, connections can be marked to be destroyed and should not be reused. Although the bad connection is removed, a following match can cause it to try to be reused, throwing the following exception:

Reviewer: ```suggestion
A `java.lang.IndexOutOfBoundsException` might occur during a backend resource outage. During an outage, connections that are marked to be destroyed should not be reused. Although the bad connection is removed, a following match can cause it to try to be reused, throwing the following exception:
```

----
+
The issue has been resolved and the `java.lang.IndexOutOfBoundsException` no longer occurs.

* link:https://github.com/OpenLiberty/open-liberty/issues/23690[JTOpen Toolbox driver 11.1 JDBC connections fail from Open Liberty to IBM i]
+
When using the IBM JTOpen toolbox driver 11.1 to establish a JDBC connection with an IBM i system, the connection fails with the following error:
+
[source]
----
{exception=The application server rejected the connection. (Password is not set.) DSRA0010E: SQL State = 08004, Error Code = -99,999, id=jdbc/database}
----
+
and stacktrace:
+
[source]
----
[INFO] [WARNING ] DSRA8021W: Warning: error setting 'password'=******: java.lang.NoSuchMethodException: [C.<init>(java.lang.String)
[INFO]  at java.base/java.lang.Class.getConstructor0(Class.java:3585)
[INFO]  at java.base/java.lang.Class.getConstructor(Class.java:2271)
[INFO]  at com.ibm.ws.jdbc.internal.JDBCDriverService.setProperty(JDBCDriverService.java:1005)
...
----
+
The runtime has been updated to handle `char[]` values for passwords, resolving the issue.

* link:https://github.com/OpenLiberty/open-liberty/issues/23582[Messaging client hangs during shutdown]
+

> There exists a small timing window when an application may try to connect to a remote messaging engine while the server-acting-as-client is creating a new keystore. During the shutdown of the messaging client a new outbound connection can be created that is then never cleaned up. The next server shutdown then hangs indefinitely.

Reviewer: ```suggestion
There exists a small timing window when an application might try to connect to a remote messaging engine while the server-acting-as-client is creating a new keystore. During the shutdown of the messaging client, a new outbound connection might be created that is never cleaned up. The next server shutdown then hangs indefinitely.
```

+

> The issue has been resolved by ensuring that no new outbound connection should be allowed while the messaging client is shutting down.

Reviewer: ```suggestion
The issue has been resolved by ensuring that no new outbound connection is allowed while the messaging client is shutting down.
```

* link:https://github.com/OpenLiberty/open-liberty/issues/23425[A syntax error in JSP compile should consistently output error JSPG0077E ]
+
The `JSPG0077E` error message was not being output to `messages.log` and `console.log` in all cases of a JSP Syntax Error (`JspCoreException`).  This message was missing if the syntax error is hit in a JSP that previously wasn't compiled and precompile is disabled.
+
The issue has been resolved and the `JSPG0077E` error message gets correctly printed in the log.

* link:https://github.com/OpenLiberty/open-liberty/issues/23392[Stopping liberty Windows service immediately after starting results in hang condition]
+

> When a Liberty server is registered as a Windows service, if the service is stopped immediately after being started, this will result in a hang condition. 

Reviewer: ```suggestion
When a Liberty server is registered as a Windows service, if the service is stopped immediately after being started, a hang condition occurs. 
```

+

> The issue has been resolved and the server no longer hands when the service is immediately stopped after being started.

Reviewer: ```suggestion
The issue has been resolved and the server no longer hangs when the service is immediately stopped after being started.
```
